provider/src/main/kotlin/org/gradle/kotlin/dsl/resolver/KotlinBuildScriptDependenciesResolver.kt

```diff
@@ -219,7 +219,9 @@ class KotlinBuildScriptDependencies(
 internal
 fun projectRootOf(scriptFile: File, importedProjectRoot: File): File {
 
-    fun isProjectRoot(dir: File) = File(dir, "settings.gradle").isFile
+    // TODO:pm remove hardcoded reference to settings.gradle
```

Comment:
:o: Maybe use `BuildLayoutFactory` as a service here?

Reply:
I think we should not use an internal Gradle service here. And this code runs in IntelliJ.
See my comment on 3035e37457f2fb0cf72fe8efe8c4419e5319188d